app/controllers/developer_controller.rb

```diff
@@ -13,7 +13,28 @@ def update_sites
 
   def production_log
     @log = if params[:grep].present?
-             `tail -n 10000 log/production.log | grep -E '#{params[:grep]}' -C #{params[:context]} --color=never`
+             grep_str = if params[:grep].instance_of?(String)
+                          params[:grep]
+                        elsif params[:grep].respond_to?(to_s)
+                          params[:grep].to_s
+                        else
+                          flash[:warning] = 'Coercion failure on params[:grep]'
+                          redirect_to(root_path) && return
+                        end
+             context_int = if params[:context].instance_of?(Integer)
+                             params[:context]
+                           elsif params[:context].respond_to?(to_i)
+                             params[:context].to_i
+                           else
+                             flash[:warning] = 'Coercion failure on params[:context]'
+                             redirect_to(root_path) && return
+                           end
+             # Ensure are of correct types in case .to_*() get overrided.
+             unless grep_str.instance_of?(String) && context_int.instance_of?(Integer)
```

Comment:
If you're doing all of that error checking above, you don't need to duplicate it here